src/main/scala/org/constellation/API.scala

```diff
@@ -222,7 +222,7 @@ class API()(implicit system: ActorSystem, val timeout: Timeout, val dao: DAO)
               }
           } ~
           path("messageService" / Segment) { channelId =>
-            complete(dao.messageService.getSync(channelId))
+            complete(dao.messageService.memPool.getSync(channelId))
```

Comment:
imo instead of `memPool.getSync` there should be `lookup` - here and above as well

Reply:
I agree -- ideally services shouldn't be leaking internals, they should expose an API surface area.